src/services/project-directory-finder.js

```diff
@@ -0,0 +1,47 @@
+class ProjectDirectoryFinder {
+  constructor({ configStore, path }) {
+    this.pathSchemaDir = configStore?.lianaOptions?.schemaDir;
+    this.path = path;
+    this.dirname = __dirname;
+
+    // NOTICE: Order does matter as packages install via yarn 2 Plug n Play mode also
+    //         include node_modules in path.
+    this.PATHS_ROOT_PACKAGES = [
+      // Yarn 2 Plug n Play mode
+      this.path.join('.yarn', 'cache'),
+      this.path.join('.yarn', 'unplugged'),
+      // Usual Yarn / NPM
+      'node_modules',
+    ];
+  }
+
+  ensureAbsolutePath(subPathsToProject) {
+    // NOTICE: on POSIX system, empty path created by previous split is skipped by path.join.
+    if (!this.path.isAbsolute(this.path.join(...subPathsToProject))) {
+      return this.path.join(this.path.sep, ...subPathsToProject);
+    }
+    return this.path.join(...subPathsToProject);
+  }
+
+  getAbsolutePath() {
+    if (this.pathSchemaDir) {
+      return this.ensureAbsolutePath(this.pathSchemaDir.split(this.path.sep));
+    }
```

Comment:
no longer used

Reply:
yes but something will be missing in theory.